### PR TITLE
feat(output) add validators for output, output.filename & output.path

### PR DESCRIPTION
**Pull request by nyrosmith**
Added basic validators for output, output.filename & output.path.
@kentcdodds Hope this is ok how it is done...

### CODE REVIEW

File: src/validators/output/index.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
-import {noop} from 'lodash'
+import {isPlainObject, noop} from 'lodash'
+import {isAbsolute} from 'path'
+
 export default [
-  {key: 'output.filename', validate: noop},
-  {key: 'output.path', validate: noop},
+  {key: 'output', validate: validateOutput},
```
Comment (kentcdodds) [inline src/validators/output/index.js]: One problem with adding a validator on the `output` object is we lose the ability to tell whether someone's provided a property that's invalid. See [this](https://github.com/kentcdodds/configuration-validator/blob/kentcdodds-patch-1/README.md#uncovered-path-validation) for more info on why that is and why it matters :-)

File: src/validators/output/index.test.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import test from 'ava'
+import outputValidator from '.'
+
+const absolutePath = process.platform === 'win32' ? 'C://foo/bar/baz' : '/foo/bar/baz'
+const relativePath = 'foo/bar'
+
+
+outputValidator.forEach((item) => {
```
Comment (kentcdodds) [inline src/validators/output/index.test.js]: Maybe we could make this simpler by exporting each validator above like:

```javascript
// output/index.js
export function validateOutputPath(value) {
  //...
}

// output/index.test.js
import * as validators from '.'

runOutputPathTests(validators.validateOutputPath)
```

Thoughts?

**Comment by kentcdodds** [comment]
Other than the two notes, this looks great! Thanks!

**Comment by nyrosmith** [comment]
I'll make the changes tomorrow. I forgot my Notebook in the office

**Comment by kentcdodds** [comment]
Sounds great. Thanks!

**Comment by nyrosmith** [comment]
@kentcdodds  Updated to your suggestions

**Comment by kentcdodds** [comment]
This looks awesome!

**Pull request merged**